engines/mock/mockengine.go

```diff
@@ -7,19 +7,22 @@ import (
 
 	"github.com/taskcluster/taskcluster-worker/engines"
 	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	log "github.com/taskcluster/taskcluster-worker/log"
 	"github.com/taskcluster/taskcluster-worker/runtime"
 )
 
 type engine struct {
 	engines.EngineBase
+	Log log.Logger
 }
 
 func init() {
 	// Register the mock engine as an import side-effect
 	extpoints.EngineProviders.Register(func(
-		options extpoints.EngineOptions,
+		options *extpoints.EngineOptions,
 	) (engines.Engine, error) {
-		return engine{}, nil
+		logger := options.Environment.Log.WithField("engine", "mock")
```

Comment:
hmm... maybe it doesn't have to be exposed from runtime environment... because we always want to force engine and plugins to use a childlogger that was passed in using a `<...>Options` structure...